src/bin/sccache-dist/cmdline/parse.rs

```diff
@@ -0,0 +1,340 @@
+use std::{env, ffi::OsString, fmt, net::SocketAddr, path::PathBuf, str::FromStr};
+
+use anyhow::bail;
+use clap::{Arg, ArgGroup, Command as ClapCommand};
+use sccache::{config, dist::ServerId};
+use syslog::Facility;
+
+use crate::cmdline::{AuthSubcommand, Command};
+
+#[derive(Debug)]
+struct TokenLength(usize);
+
+impl TokenLength {
+    fn as_bytes(&self) -> usize {
+        self.0 / 8
+    }
+
+    fn from_bits(bits: &str) -> anyhow::Result<Self> {
+        let bits: usize = bits.parse()?;
+        if bits % 8 != 0 || bits < 64 || bits > 4_096 {
```

Comment:
```suggestion
        if bits & 0x7 != 0 || bits < 64 || bits > 4_096 {
```

I'd prefer to have separate error messages for each of these

Reply:
Resolved in 4b9d5bd58e3b0f9fa59de73815b68c0f9ab3be02